api/cluster/EnvironmentRestHandler.go

```diff
@@ -289,3 +290,44 @@ func (impl EnvironmentRestHandlerImpl) GetEnvironmentListForAutocomplete(w http.
 	}
 	common.WriteJsonResp(w, err, grantedEnvironment, http.StatusOK)
 }
+
+func (impl EnvironmentRestHandlerImpl) GetCombinedEnvironmentListForDropDown(w http.ResponseWriter, r *http.Request) {
+	userId, err := impl.userService.GetLoggedInUser(r)
+	if userId == 0 || err != nil {
+		common.WriteJsonResp(w, err, "Unauthorized User", http.StatusUnauthorized)
+		return
+	}
+
+	environments, err := impl.environmentClusterMappingsService.GetCombinedEnvironmentListForDropDown()
+	if err != nil {
+		impl.logger.Errorw("service err, GetCombinedEnvironmentListForDropDown", "err", err)
+		common.WriteJsonResp(w, err, nil, http.StatusInternalServerError)
+		return
+	}
+
+	token := r.Header.Get("token")
+	// RBAC enforcer applying
+	var clusters []request.ClusterEnvDto
+	grantedEnvironmentMap := make(map[string][]*request.EnvDto)
+	for _, environment := range environments {
+		if ok := impl.enforcer.Enforce(token, casbin.ResourceGlobalEnvironment, casbin.ActionGet, strings.ToLower(environment.Environment)); ok {
+			grantedEnvironmentMap[environment.ClusterName] = append(grantedEnvironmentMap[environment.ClusterName], &request.EnvDto{
+				EnvironmentId:   environment.Id,
+				EnvironmentName: environment.Environment,
+				Namespace:       environment.Namespace,
+			})
+		}
+	}
+	//RBAC enforcer Ends
+
+	for k, v := range grantedEnvironmentMap {
```

Comment:
can we merge this in previous for loop

Reply:
no, grouping is separated from auth enforcer.